### PR TITLE
Create "Step" and "Info" type for JSON output for minikube start

### PR DESCRIPTION
**Pull request by priyawadhwa**
This PR begins implementing JSON output for minikube start

It only converts regular steps to JSON -- warnings, errors and downloads will come in later PRs.

This PR also adds integration tests to make sure:
1. the output is cloud events compatible
1. each log has a unique step number

Output currently looks like this (assuming all artifacts have already been downloaded):

```
$ minikube start --output json

{"data":{"currentstep":"0","message":"minikube v1.12.0-beta.1 on Darwin 10.14.6\n","name":"Initial Minikube Setup","totalsteps":"9"},"datacontenttype":"application/json","id":"37d6fc67-1202-4155-9f0a-05c673cc55cc","source":"https://minikube.sigs.k8s.io/","specversion":"1.0","type":"io.k8s.sigs.minikube.step"}
{"data":{"currentstep":"1","message":"Automatically selected the docker driver. Other choices: hyperkit, virtualbox\n","name":"Selecting Driver","totalsteps":"9"},"datacontenttype":"application/json","id":"bc783fae-e93c-4ae2-83ae-e814ae1e19e4","source":"https://minikube.sigs.k8s.io/","specversion":"1.0","type":"io.k8s.sigs.minikube.step"}
{"data":{"currentstep":"3","message":"Starting control plane node minikube in cluster minikube\n","name":"Starting Node","totalsteps":"9"},"datacontenttype":"application/json","id":"9b98194d-15c0-4e25-8620-5325f7ba8af8","source":"https://minikube.sigs.k8s.io/","specversion":"1.0","type":"io.k8s.sigs.minikube.step"}
{"data":{"currentstep":"4","message":"Creating docker container (CPUs=2, Memory=1988MB) ...\n","name":"Creating Container","totalsteps":"9"},"datacontenttype":"application/json","id":"a9d3e271-2cb6-4ae7-ba5b-7123f09ba539","source":"https://minikube.sigs.k8s.io/","specversion":"1.0","type":"io.k8s.sigs.minikube.step"}
{"data":{"currentstep":"6","message":"Preparing Kubernetes v1.18.3 on Docker 19.03.2 ...\n","name":"Preparing Kubernetes","totalsteps":"9"},"datacontenttype":"application/json","id":"39dd5d7b-abfd-46d1-8c25-026c34497f4a","source":"https://minikube.sigs.k8s.io/","specversion":"1.0","type":"io.k8s.sigs.minikube.step"}
{"data":{"currentstep":"7","message":"Verifying Kubernetes components...\n","name":"Verifying Kubernetes","totalsteps":"9"},"datacontenttype":"application/json","id":"13c372ef-5fc4-4049-a949-e31b06f24272","source":"https://minikube.sigs.k8s.io/","specversion":"1.0","type":"io.k8s.sigs.minikube.step"}
{"data":{"currentstep":"8","message":"Enabled addons: default-storageclass, storage-provisioner\n","name":"Enabling Addons","totalsteps":"9"},"datacontenttype":"application/json","id":"74c132c1-de02-4302-95aa-501d2cc56ca9","source":"https://minikube.sigs.k8s.io/","specversion":"1.0","type":"io.k8s.sigs.minikube.step"}
{"data":{"currentstep":"9","message":"Done! kubectl is now configured to use \"minikube\"\n","name":"Done","totalsteps":"9"},"datacontenttype":"application/json","id":"29c07ccd-61a4-4f80-8931-5a57b51d58b3","source":"https://minikube.sigs.k8s.io/","specversion":"1.0","type":"io.k8s.sigs.minikube.step"}
```

#8655

### CODE REVIEW

**Comment by medyagh** [comment]
Docker Driver flakes: TestNetworkPlugins/group/false/DNS  
- https://github.com/kubernetes/minikube/issues/8665

File: cmd/minikube/cmd/start_flags.go [hunk 1]
```diff
@@ -144,6 +145,7 @@ func initMinikubeFlags() {
 	startCmd.Flags().Bool(preload, true, "If set, download tarball of preloaded images if available to improve start time. Defaults to true.")
 	startCmd.Flags().Bool(deleteOnFailure, false, "If set, delete the current cluster if start fails and try again. Defaults to false.")
 	startCmd.Flags().Bool(forceSystemd, false, "If set, force the container runtime to use sytemd as cgroup manager. Currently available for docker and crio. Defaults to false.")
+	startCmd.Flags().String(startOutput, "", "Format to print stdout in. Options include: [json]")
```
Comment (medyagh) [inline cmd/minikube/cmd/start_flags.go:148]: we should have a name for the default option . simmilar to other cli tools

Options include: text, json

and default is text

Comment (medyagh) [reply cmd/minikube/cmd/start_flags.go:148]: for example az 

    --output -o         : Output format.  Allowed values: json, jsonc, none, table, tsv, yaml,
                          yamlc.  Default: json.

Comment (priyawadhwa) [reply cmd/minikube/cmd/start_flags.go]: done

**Comment by priyawadhwa** [comment]
this flake occurs pretty consistently across platforms

https://github.com/kubernetes/minikube/issues/8666

**Pull request merged**